graylog2-server/src/main/java/org/graylog2/buffers/processors/ServerProcessBufferProcessor.java

```diff
@@ -41,10 +42,10 @@
 
     @Inject
     public ServerProcessBufferProcessor(MetricRegistry metricRegistry,
-                                        OrderedMessageProcessors orderedMessageProcessors,
+                                        Provider<OrderedMessageProcessors> orderedMessageProcessorsProvider,
                                         OutputBuffer outputBuffer) {
         super(metricRegistry);
-        this.orderedMessageProcessors = orderedMessageProcessors;
+        this.orderedMessageProcessors = orderedMessageProcessorsProvider.get();
```

Comment:
The thing is that a (singleton) provider could also return always the same instance of the injectee and it doesn't help clarifying the contract (that we need a new instance every time).